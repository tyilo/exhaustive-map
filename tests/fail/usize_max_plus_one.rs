//@normalize-stderr-test: "'\S*+long-type-\d+.txt" -> "'long-type-n.txt"

use exhaustive_map::Finite;
use std::num::NonZeroUsize;

#[derive(Finite)]
enum TooBig {
    A(NonZeroUsize),
    B,
}
