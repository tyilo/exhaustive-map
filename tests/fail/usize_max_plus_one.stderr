error[E0271]: type mismatch resolving `<UInt<UInt<..., ...>, ...> as IsLessOrEqual<...>>::Output == B1`
 --> tests/fail/usize_max_plus_one.rs:6:10
  |
6 | #[derive(Finite)]
  |          ^^^^^^ expected `B1`, found `B0`
  |
  = note: required for `UInt<UInt<UInt<UInt<UInt<UInt<..., ...>, ...>, ...>, ...>, ...>, ...>` to implement `exhaustive_map::finite::sealed::Sealed`
  = help: see issue #48214
  = note: the full name for the type has been written to 'long-type-n.txt'
  = note: consider using `--verbose` to print the full type name to the console
  = note: this error originates in the derive macro `Finite` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0271`.
